plugins/module_utils/config_utils.py

```diff
@@ -99,3 +134,370 @@ def changed(self) -> bool:
+        dictionary of nested settings if 'interface' is a nested setting within 'network'.
+
+        Note:
+        - The function emphasizes the importance of a properly initialized `version_map`.The absence
+        of the module in the map or the setting within the module leads to a return value of None.
+        """
+
+        if not self.version_map:
+            raise OPNSenseConfigUsageError(
+                "Module specific version_map was not set during initalization"
+            )
+
+        map_dict = self._get_module_version_map(module=module)
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        raise OPNSenseConfigUsageError(
+            "Module specific xpath was not found"
+        )
+
+    def _get_php_requirements(self, module: str, setting: str) -> list:
+        """
+        Retrieves a list of PHP requirements for a specific module and setting based on the
+        version-specific mapping.
+
+        This function looks up PHP requirements in the version-specific mapping for the given
+        module and setting. It returns the list associated with the setting if it's directly
+        under the module. If the setting is nested, it performs a recursive search.
+
+        Parameters:
+        - module (str): The module's name for which PHP requirements are needed.
+        - setting (str): The specific setting within the module.
+
+        Returns:
+        - Optional[list]: List of PHP requirements if found, or None if not found or unavailable.
+
+        Example:
+        - Calling _get_php_requirements('network', 'extensions') might return a list of required
+          PHP extensions for the 'network' module, if 'extensions' is a setting under 'network'.
+
+        Note:
+        - The accuracy of the returned requirements depends on the completeness of `version_map`.
+          If a module or setting is missing in the map, the function returns None.
+        """
+
+        map_dict = self._get_module_version_map(module=module)
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        raise OPNSenseConfigUsageError(
+            "Module specific get_php_requirement were not found"
+        )
+
+    def _get_configure_functions(self, module: str, setting: str) -> dict:
```

Comment:
Here as well. Is the settings argument necessary? If we want to access _get_configure_functions we know we will access the `configure_functions` in the VERSION_MAP

Reply:
Same as in `_get_php_requirements`.As stated in the docstring, the functions are looked up depending on the setting and can therefore not be hardcoded in the function.